src/Symfony/Component/Lock/Store/PdoStore.php

```diff
@@ -127,8 +127,19 @@ public function save(Key $key)
         try {
             $stmt->execute();
         } catch (\PDOException $e) {
-            // the lock is already acquired. It could be us. Let's try to put off.
-            $this->putOffExpiration($key, $this->initialTtl);
+            if ($this->isTableMissing($e) && (!$conn->inTransaction() || \in_array($this->driver, ['pgsql', 'sqlite', 'sqlsrv'], true))) {
```

Comment:
I just noticed that the `PdoAdapter` has [the same logic](https://github.com/symfony/symfony/blob/5725df0b0dae86a5e7cbc30493f4c900412b7882/src/Symfony/Component/Cache/Adapter/PdoAdapter.php#L321-L328) but without the `$this->isTableMissing()` part so I'm wondering, should I add it there as well, or remove it form here to keep it consistent?

Reply:
yes indeed, also in DoctrineDbalAdapter, right?

Reply:
The DBAL adapter catches TableNotFoundException so it is already fine (and the DBAL adapter uses the DBAL API that does the preparation and the execution internally so the try/catch wraps both).

Reply:
the try/catch is around the call to prepare, not to execute, that's why I'm wondering

Reply:
I've added the logic to `PdoAdapter`, the `DoctrineDbalAdapter` already works as expected. I've added some tests to the adapters/stores to make super everything works as expected.